src/process.rs

```diff
@@ -404,22 +510,39 @@ mod unix {
     #[derive(Clone, Debug, Deserialize, Serialize)]
     #[serde(try_from = "String", into = "String", expecting = "a user name")]
     struct UserId {
-        /// The numerical user ID.
-        uid: Uid,
-
         /// The user name.
         ///
-        /// We keep this information so we can produce the actual config.
+        /// This is used for error reporting.
         name: String,
+
+        /// The user name as a C string.
+        ///
+        /// This is used internally. We keep both the string and C string
+        /// versions because conversion can cause errors and we want to catch
```

Comment:
How does keeping both versions allow you to catch errors early? This comment is not clear to me.